System.Containers.Tasks/CreateNewImage.cs

```diff
@@ -1,105 +1,146 @@
-        /// 1. Pull a base image (needs parameters: URL, BaseImage, BaseImageTag)
-        /// 2. Add output of build as a new layer
-        /// 3. Push image back to some registry (needs parameters: OutputURL, NewName, EntryPoint)
-        /// </summary>
-        /// <returns></returns>
-        public override bool Execute()
+        BaseRegistry = "";
+        BaseImageName = "";
+        BaseImageTag = "";
+        OutputRegistry = "";
+        ImageName = "";
+        ImageTag = "";
+        PublishDirectory = "";
+        WorkingDirectory = "";
+        Entrypoint = "";
+        EntrypointArgs = "";
+    }
+
+
+    public override bool Execute()
+    {
+        if (string.IsNullOrEmpty(PublishDirectory))
         {
-            if (string.IsNullOrEmpty(PublishDirectory) || !Directory.Exists(PublishDirectory))
-            {
-                Log.LogError("PublishDirectory and Files are both invalid. One valid parameter MUST be given to the CreateNewImage task.");
-                return false;
-            }
+            Log.LogError(string.Format("PublishDirectory is must have a value"));
+            return !Log.HasLoggedErrors;
+        }
+        else if (!Directory.Exists(PublishDirectory))
+        {
+            Log.LogError(string.Format("PublishDirectory does not exist."));
+            return !Log.HasLoggedErrors;
+        }
 
-            Registry reg = new Registry(new Uri(InputRegistryURL));
+        Registry reg;
+        Image image;
 
-            Image image;
-            try
-            {
-                image = reg.GetImageManifest(BaseImageName, BaseImageTag).Result;
-            }
-            catch (Exception ex)
+        try
+        {
+            reg = new Registry(new Uri(BaseRegistry, UriKind.RelativeOrAbsolute));
+            image = reg.GetImageManifest(BaseImageName, BaseImageTag).Result;
+        }
+        catch (Exception e)
+        {
+            if (BuildEngine != null)
             {
-                Log.LogError("GetImageManifest Failed: {0}.\n{1}", ex.Message, ex.InnerException);
-                return false;
+                Log.LogError("Failed getting image manifest: {0}", e);
             }
+            return !Log.HasLoggedErrors;
+        }
 
+        if (BuildEngine != null)
+        {
             Log.LogMessage($"Loading from directory: {PublishDirectory}");
-            Layer newLayer = Layer.FromDirectory(PublishDirectory, WorkingDirectory);
-            image.AddLayer(newLayer);
-
-            image.SetEntrypoint(Entrypoint, EntrypointArgs?.Split(' ').ToArray());
-
-            Registry outputReg = new Registry(new Uri(OutputRegistryURL));
+        }
+        
+        Layer newLayer = Layer.FromDirectory(PublishDirectory, WorkingDirectory);
+        image.AddLayer(newLayer);
+        image.SetEntrypoint(Entrypoint, EntrypointArgs?.Split(' ').ToArray());
 
+        if (OutputRegistry.StartsWith("docker://"))
+        {
+            // To Do: LocalDocker.Load();
```

Comment:
this is where @rainersigwald's new docker feature plugs in, yeah? We'll want to get that in for the initial preview.

Reply:
Yep! It's on the shortlist